website/statistics.py

```diff
@@ -122,6 +126,610 @@ def get_program_stats(self, user):
         return jsonify(response)
 
 
+class LiveStatisticsModule(WebsiteModule):
+    def __init__(self, db: Database):
+        super().__init__("live-stats", __name__)
+        self.db = db
+        # Define the groups of misconceptions
+        self.misconception_groups = {
+            'Not a current level command': ['level'],
+            'Incorrect use of command': ['cannot'],
+            'Incorrect use of variable': ['variable'],
+            'Unwanted spaces': ['Spaces', 'confuse', 'computers'],
+            'Forgot commandos': ['forgot'],
+            'Empty program': ['empty program'],
+            'Typed something that is not allowed': ['entered', 'allowed'],
+            'Echo and ask mismatch': ['echo before an ask', 'echo without an ask'],
+        }
+        self.MAX_CONTINUOUS_ERRORS = 3
+        self.MAX_COMMON_ERRORS = 10
+        self.MAX_FEED_SIZE = 4
+
+    def __selected_levels(self, class_id):
+        class_customization = self.db.get_class_customizations(class_id)
+        class_overview = class_customization.get('dashboard_customization')
+        if class_overview:
+            return class_overview.get('selected_levels', [1])
+        return [1]
+
+    def __common_errors(self, class_id):
+        class_customization = self.db.get_class_customizations(class_id)
+        # if the class customization button has not been pressed yet
+        # class_customization will be None
+        # if that is the case, we create a default class customization
+        if not class_customization:
+            sorted_adventures = {}
+            for lvl, adventures in hedy_content.ADVENTURE_ORDER_PER_LEVEL.items():
+                sorted_adventures[str(lvl)] = [{'name': adventure, 'from_teacher': False} for adventure in adventures]
+            customizations = {
+                "id": class_id,
+                "levels": [i for i in range(1, HEDY_MAX_LEVEL + 1)],
+                "opening_dates": {},
+                "other_settings": [],
+                "level_thresholds": {},
+                "sorted_adventures": sorted_adventures
+            }
+            self.db.update_class_customizations(customizations)
+            class_customization = self.db.get_class_customizations(class_id)
```

Comment:
If you are saving the customizations object you don't need to to get it from the database. This way you can save it right away with the field `dashboard_customization`